data/transactions/logic/evalCrypto_test.go

```diff
@@ -173,6 +175,154 @@ ed25519verify_bare`, pkStr), v)
+	*/
+	// generated by github.com/p-steuer/eddsa-test
+	data, err := hex.DecodeString("cb68")
+	require.NoError(t, err)
+	vecs := [][3]string{
+		{
+			"50726f67446174610fefbfeb4749642b4e93f37098946ce11a63d6a8db82a416ff91321760212faccb68",                                             // msg M
+			"5866666666666666666666666666666666666666666666666666666666666666",                                                                 // pub A
+			"01000000000000000000000000000000000000000000000000000000000000006a889c732ac8ea1507a1547dcee1d74efc9113f9377216d8dfacc7fee5156504", // sig (R,S)
+		},
+	}
+	vecBytes := make([][3][]byte, len(vecs))
+	for i, tup3 := range vecs {
+		for j := range tup3 {
+			vecBytes[i][j], err = hex.DecodeString(vecs[i][j])
+			require.NoError(t, err)
+		}
+	}
+
+	for i, vec := range vecBytes {
+		t.Run(fmt.Sprintf("valid,vec=%v", vecs[i]), func(t *testing.T) {
+			// simple program that is only valid if EnableBatchVerification=true
+			ops, err := AssembleStringWithVersion(fmt.Sprintf(`arg 0
+arg 1
+arg 2
+ed25519verify`), uint64(6))
+			require.NoError(t, err)
+			var txn transactions.SignedTxn
+			txn.Lsig.Logic = ops.Program
+			txn.Lsig.Args = [][]byte{data[:], vec[2], vec[1]}
+
+			// try with EnableBatchVerification = false
+			evalParams := defaultEvalParams(&txn)
+			evalParams.Proto.EnableBatchVerification = false
+			pass, err := EvalSignature(0, evalParams)
+			require.False(t, pass)
+			require.NoError(t, err)
+
+			// try with EnableBatchVerification = true
+			evalParams.Proto.EnableBatchVerification = true
+			pass, err = EvalSignature(0, evalParams)
+			require.True(t, pass)
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestEd25519verifyCompatPayment(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	// generated by github.com/p-steuer/eddsa-test
+	data, err := hex.DecodeString("1dab")
+	require.NoError(t, err)
+	vecs := [][3]string{
+		{
+			"50726f6744617461f6691f135f7945d27ec1f87846d0694709e2ae67808ff4082a007a15754cb9a51dab",                                             // msg M
+			"5866666666666666666666666666666666666666666666666666666666666666",                                                                 // pub A
+			"0100000000000000000000000000000000000000000000000000000000000000095660bba17daf63dac3dc8c8600c55e25dd077032ff3c050bb0e9cdd0293b0e", // sig (R,S)
+		},
+	}
+	vecBytes := make([][3][]byte, len(vecs))
+	for i, tup3 := range vecs {
+		for j := range tup3 {
+			vecBytes[i][j], err = hex.DecodeString(vecs[i][j])
+			require.NoError(t, err)
+		}
+	}
+
+	for i, vec := range vecBytes {
+		t.Run(fmt.Sprintf("valid,vec=%v", vecs[i]), func(t *testing.T) {
+			// logic sig that lets you withdraw different amounts if EnableBatchVerification=true
+			/* (||
+			    (&& (ed25519verify (arg 0) (arg 1) (arg 2))
+			      (== (txn Amount) 1000000))
+			    (&& (! (ed25519verify (arg 0) (arg 1) (arg 2)))
+			      (== (txn Amount) 2000000))
+			   )
+			*/
+			ops, err := AssembleStringWithVersion(`arg 0
```

Comment:
I like to use `testProg` for assembling in tests. It checks the error for you and does not return it, it disassembles and reassembles to check equality giving us more tests for free, and does some logging if things don't work out.